pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -196,6 +196,12 @@ func (rs *RevisionStatus) MarkResourcesUnavailable(reason, message string) {
 	revCondSet.Manage(rs).MarkFalse(RevisionConditionResourcesAvailable, reason, message)
 }
 
+// MarkSecretUnavailable changes "ResourcesAvailable" condition to false to reflect that there
+// is an error getting the secret by name.
+func (rs *RevisionStatus) MarkSecretUnavailable(secretName, message string) {
```

Comment:
The number of functions here is starting to get unwieldy.  I don't think we need to change it for this, but created an issue to clean up: https://github.com/knative/serving/issues/5780